src/components/Firestore/Collection.test.tsx

```diff
@@ -227,9 +227,13 @@ it('shows the selected sub-document', async () => {
 });
 
 describe('withCollectionState', () => {
-  let performAddDocument: () => Promise<void>;
+  let performAddDocument: (hasSpecialId?: boolean) => Promise<void>;
```

Comment:
nit: Boolean arguments are hard to read in general -- consider changing it to `performAddDocument(id = 'new-document-id')` and callers can pass in any id (including special characters) if they want to.